odoo/models.py

```diff
@@ -178,6 +173,9 @@ def __new__(meta, name, bases, attrs):
     def __init__(self, name, bases, attrs):
         super().__init__(name, bases, attrs)
 
+        if attrs.get('_execute_sql'):
+            _logger.warning("Deprecated method %s._execute_sql in module %s", name, attrs.get('__module__'))
+
```

Comment:
Might be a good idea to also trigger warnings for `__init__` as its behaviour has changed? If the signature doesn't match the new one anyway (aka if it doesn't have 4 parameters).

It's going to crash anyway but being forewarned with a clear message is helpful.

Reply:
Yep, it is legit.

Reply:
Obviously there is a model who override `__init__` (`ir.ui.menu`) (then thanks). The code seems completely depreciated (`register_cache_clearing_method`), I am going to check that crap.